multiaddr_test.go

```diff
@@ -923,3 +924,36 @@ func TestDNS(t *testing.T) {
 		t.Fatal("expected equality")
 	}
 }
+
+func TestHTTPPath(t *testing.T) {
+	t.Run("bad addr", func(t *testing.T) {
+		badAddr := "/http-path/thisIsMissingAfullBytes%f"
+		_, err := NewMultiaddr(badAddr)
+		require.Error(t, err)
+	})
+
+	t.Run("round trip", func(t *testing.T) {
+		cases := []string{
+			"/http-path/tmp%2Fbar",
+			"/http-path/tmp%2Fbar%2Fbaz",
+			"/http-path/foo",
```

Comment:
Can you add a case with the complete multiaddr with ipv4 + p2p + http + http-path or just ipv4 + p2p + http-path?